test/onnx/gen_onnx.py

```diff
@@ -2462,6 +3014,111 @@ def if_pl_test():
     return ([node], [cond_input, x, y], [ret], [xt_tensor, yt_tensor])
 
 
+@onnx_test
+def if_else_trailing_one_shape_test():
+    x = onnx.helper.make_tensor_value_info('x', onnx.TensorProto.FLOAT, [2, 1])
+    y = onnx.helper.make_tensor_value_info('y', onnx.TensorProto.FLOAT, [2])
+
+    then_out = onnx.helper.make_tensor_value_info('then_out',
+                                                  onnx.TensorProto.FLOAT,
+                                                  [2, 1])
+    else_out = onnx.helper.make_tensor_value_info('else_out',
+                                                  onnx.TensorProto.FLOAT, [2])
+
+    xt = np.ones((2, 1)).astype(np.float)
+    xt_tensor = helper.make_tensor(name='xt',
+                                   data_type=TensorProto.FLOAT,
+                                   dims=xt.shape,
+                                   vals=xt.flatten().astype(np.float32))
+
+    yt = np.random.randn(2).astype(np.float)
+    yt_tensor = helper.make_tensor(name='yt',
+                                   data_type=TensorProto.FLOAT,
+                                   dims=yt.shape,
+                                   vals=yt.flatten().astype(np.float32))
+
+    then_add_node = onnx.helper.make_node('Add',
+                                          inputs=['x', 'xt'],
+                                          outputs=['then_out'])
+
+    else_mul_node = onnx.helper.make_node('Mul',
+                                          inputs=['y', 'yt'],
+                                          outputs=['else_out'])
+
+    then_body = onnx.helper.make_graph([then_add_node], 'then_body', [],
+                                       [then_out])
+
+    else_body = onnx.helper.make_graph([else_mul_node], 'else_body', [],
+                                       [else_out])
+
+    cond = np.array([0]).astype(np.bool)
+    cond_tensor = helper.make_tensor(name="cond",
+                                     data_type=TensorProto.BOOL,
+                                     dims=cond.shape,
+                                     vals=cond.astype(bool))
+    res = onnx.helper.make_tensor_value_info('res', TensorProto.FLOAT, [])
+
+    node = onnx.helper.make_node('If',
+                                 inputs=['cond'],
+                                 outputs=['res'],
+                                 then_branch=then_body,
+                                 else_branch=else_body)
+
+    return ([node], [x, y], [res], [cond_tensor, xt_tensor, yt_tensor])
+
+
+def if_then_empty_shape_test():
```

Comment:
Would it be possible to add tests for multiple outputs from then and else branch ?

Reply:
I have one already to test differing shapes on output. You're saying adding more? Is there a particular test you want me to add?

Reply:
I see this one : if_then_else_incompatible_output_shapes_test 
Without mutiple outputs from then and else branch. But it is testing for incompatibility. Can we have similar test but with compatible shapes ?

Reply:
Sure I can add a larger tensor and try that as well and retest the trailing case too

Reply:
> Sure I can add a larger tensor and try that as well and retest the trailing case too

Why larger tensor ? we want to test multiple output from both `then` and `else` branch, and shapes of which should be compatible.

Reply:
Just added two tests for that, if_then_else_test_multi_outputs_test, if_then_else_test_multi_outputs_test2 in onnx_test for that. 

I've got the last case I'm working on for the empty multi output vector I'm gonna add as well.